python/hsfs/core/feature_view_engine.py

```diff
@@ -923,3 +935,138 @@ def _check_if_exists_with_prefix(self, f_name, f_set):
             )
         else:
             return f_name
+
+    def enable_feature_logging(self, fv):
+        self._feature_view_api.enable_feature_logging(fv.name, fv.version)
+        fv.enabled_logging = self.get(fv.name, fv.version).enabled_logging
+        return fv
+
+    def get_feature_logging(self, fv):
+        return FeatureLogging.from_response_json(
+            self._feature_view_api.get_feature_logging(fv.name, fv.version)
+        )
+
+    def _get_logging_fg(self, fv, transformed):
+        feature_logging = self.get_feature_logging(fv)
+        if transformed:
+            return feature_logging.transformed_features
+        else:
+            return feature_logging.untransformed_features
+
+    def log_features(self, fv, features, prediction=None, transformed=False, write_options=None, training_dataset_version=None, hsml_model=None):
+        default_write_options = {
+            "start_offline_materialization": False,
+        }
+        default_write_options.update(write_options)
+        fg = self._get_logging_fg(fv, transformed)
+        df = engine.get_instance().get_feature_logging_df(
+            fg,
+            features,
+            [feature for feature in fv.features if not feature.label],
+            [feature for feature in fv.features if feature.label],
+            FeatureViewEngine._LOG_TD_VERSION,
+            FeatureViewEngine._LOG_TIME,
+            FeatureViewEngine._HSML_MODEL,
+            prediction,
+            training_dataset_version,
+            hsml_model,
+        )
+        return fg.insert(df, write_options=default_write_options)
```

Comment:
Have you thought of using the `multi_insert` method to send feature vectors directly to kafka without batching them first?

Reply:
The problem is  it require users to call `finalize_multi_part_insert` otherwise the log wont be send to kafka

Reply:
We can extend this with the `multi_insert` as an extra option later. It would be nice to have if users do not forget to call `finalize_multi_part_insert`. But if there is a container to flush the feature, then it should be fine.